### ISSUE TITLE
Expected results for GAN?

### ISSUE THREAD

**Issue by fetchlister** [deepfakes/faceswap#210]
I know gan is supposed to be smoother than the default trainer but there is little information about how to go about training with gan. I hear some places say it takes 8 times longer to train on gan than n the default trainer, is this true? what loss number should I shoot for in which of the 4 categories? How long can I expect to train on a good GPU (3584 cuda cores)? 

Its all very confusing seeing my GAN stuck at a loss of 17 for days not knowing if it will take longer, or if I'm looking at the wrong number. I think documentation would help.

**Comment by andykdy** [comment]
I haven't seen the implementation of GANs in this project specifically, but designing a Generative Adversarial Network can be a pretty difficult task. 
If you reward either the Discriminator or Generator incorrectly, it can have a snowball effect where one of the agent overpowers the other and the optimization of loss stagnates. 
If I find the paper on it I will link it, but I remember it having to do with batch normalization...

**Comment by iperov** [comment]
there is bad implement of gan in faceswap :(
looks like v1 and without perceptual loss

**Comment by ppmdo** [comment]
@iperov So you mean that V2:https://github.com/shaoanlu/faceswap-GAN/blob/master/FaceSwap_GAN_v2_train.ipynb should be integrated to this codebase?

**Comment by iperov** [comment]
@ppmdo yes, with perceptual loss :)

**Comment by petergerten** [comment]
this seems to be a good start: https://github.com/deepfakes/faceswap/pull/151
But it is missing the converter yet, and I could not get the converter running myself.
